src/ToricVarieties/CohomologyClasses/constructors.jl

```diff
@@ -0,0 +1,297 @@
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> dc = ToricDivisorClass(P2, [1])
+A divisor class on a normal toric variety
+
+julia> cc = CohomologyClass(dc)
+A cohomology class on a normal toric variety given by x3
+
+julia> cc - cc
+A cohomology class on a normal toric variety given by 0
+```
+"""
+function Base.:-(cc1::CohomologyClass, cc2::CohomologyClass)
+    if toric_variety(cc1) !== toric_variety(cc2)
+        throw(ArgumentError("The cohomology classes must be defined on identically the same toric variety."))
+    end
+    ring = cohomology_ring(toric_variety(cc1))
+    poly = polynomial(cc1, ring) - polynomial(cc2, ring)
+    return CohomologyClass(toric_variety(cc1), poly)
+end
+
+
+@doc Markdown.doc"""
+    Base.:*(c::fmpq, cc::CohomologyClass)
+
+Return `c`-times the cohomology class `cc`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> tdc = ToricDivisorClass(P2, [1])
+A divisor class on a normal toric variety
+
+julia> fmpq(5//2)*CohomologyClass(tdc)
+A cohomology class on a normal toric variety given by 5//2*x3
+```
+"""
+Base.:*(c::fmpq, cc::CohomologyClass) = CohomologyClass(toric_variety(cc), coefficient_ring(toric_variety(cc))(c) * polynomial(cc))
+Base.:*(c::Rational{Int64}, cc::CohomologyClass) = CohomologyClass(toric_variety(cc), coefficient_ring(toric_variety(cc))(c) * polynomial(cc))
+Base.:*(c::fmpz, cc::CohomologyClass) = CohomologyClass(toric_variety(cc), coefficient_ring(toric_variety(cc))(c) * polynomial(cc))
+Base.:*(c::Int, cc::CohomologyClass) = CohomologyClass(toric_variety(cc), coefficient_ring(toric_variety(cc))(c) * polynomial(cc))
+
+
+#################################
+# 4: Wedge product
+#################################
+
+@doc Markdown.doc"""
+    Base.:*(cc1::CohomologyClass, cc2::CohomologyClass)
+
+Return the wedge product of the cohomology classes `cc1` and `cc2`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> dc = ToricDivisorClass(P2, [1])
+A divisor class on a normal toric variety
+
+julia> cc = CohomologyClass(dc)
+A cohomology class on a normal toric variety given by x3
+
+julia> cc * cc
+A cohomology class on a normal toric variety given by x3^2
+```
+"""
+function Base.:*(cc1::CohomologyClass, cc2::CohomologyClass)
+    if toric_variety(cc1) !== toric_variety(cc2)
+        throw(ArgumentError("The cohomology classes must be defined on identically the same toric variety."))
+    end
+    ring = cohomology_ring(toric_variety(cc1))
+    poly = polynomial(cc1, ring) * polynomial(cc2, ring)
+    return CohomologyClass(toric_variety(cc1), poly)
+end
```

Comment:
I propose the following naming convention:
- Introduce the function `wedge(cc1::CohomologyClass, cc2::CohomologyClass)`
- `Base.:*(cc1::CohomologyClass, cc2::CohomologyClass) = wedge(cc1, cc2)`

That way we can have `wedge` in the docs without cluttering the docs for `Base.*` and (to me) the code is clearer, since `wedge` is a word I know, but `*` is a little ambigous (though there may be no other operation in this case). I propose to proceed similarly for the other `Base.smth` operations too, at least if we want them in the docs.

Reply:
Thx for all the feedback. All of this sounds good and reasonable. Will try to adjust this PR shortly.

Reply:
I would prefer `*`.

Internally, the (toric) cohomology classes are polynomials in the cohomology ring. The wedge product is just the product of such polynomials. This makes  it natural to pick `*`.

It is also rather convenient (I anticipate heavy use of this). Namely, instead of `wedge(cc1, cc2)` I can just write `cc1*cc2`. It also immediately generalizes to `cc1*cc2*cc3*...`.

Regarding the documentation, I will follow the suggestion by @fingolfin above, remove the doc strings and elaborate directly in the documentation.